@ywelsch looks like we have a possible related test failure:

```
6:40:10 FAILURE 4.92s J4 | SnapshotBlocksIT.testCreateSnapshotWithIndexBlocks <<< FAILURES!
16:40:10    > Throwable #1: java.lang.AssertionError: Request executed with success but a ClusterBlockException was expected
16:40:10    > 	at __randomizedtesting.SeedInfo.seed([117D2CBB1420960:B693C3083F0F2E94]:0)
16:40:10    > 	at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertBlocked(ElasticsearchAssertions.java:166)
16:40:10    > 	at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertBlocked(ElasticsearchAssertions.java:191)
16:40:10    > 	at org.elasticsearch.action.admin.cluster.snapshots.SnapshotBlocksIT.testCreateSnapshotWithIndexBlocks(SnapshotBlocksIT.java:117)
16:40:10    > 	at java.lang.Thread.run(Thread.java:748)
```

--- a/server/src/test/java/org/elasticsearch/action/admin/cluster/snapshots/SnapshotBlocksIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/cluster/snapshots/SnapshotBlocksIT.java
@@ -113,8 +113,8 @@ public class SnapshotBlocksIT extends ESIntegTestCase {
         logger.info("-->  creating a snapshot is blocked when an index is blocked for reads");
         try {
             enableIndexBlock(INDEX_NAME, SETTING_BLOCKS_READ);
-            client().admin().cluster().prepareCreateSnapshot(REPOSITORY_NAME, "snapshot-2")
-                .setIndices(COMMON_INDEX_NAME_MASK).get();
+            assertThat(client().admin().cluster().prepareCreateSnapshot(REPOSITORY_NAME, "snapshot-2")
+                .setIndices(COMMON_INDEX_NAME_MASK).setWaitForCompletion(true).get().status(), equalTo(RestStatus.OK));
         } finally {
             disableIndexBlock(INDEX_NAME, SETTING_BLOCKS_READ);
         }

